nifi-api/src/main/java/org/apache/nifi/processor/ProcessSession.java

```diff
@@ -109,6 +109,19 @@
     void rollback(boolean penalize);
 
     /**
+     * Produces a listing of pertinent information about the session's
+     * unacknowledged flowfiles that can be used for logging purposes.  If the
+     * session fails for any reason this can help identify problem files with
+     * minimal system impact.
+     *
+     * Because of testing dependencies and ot
+     *
+     * @return {@link String} listing pertinent information about the session's
+     * unacknowledged flowfiles, primarily intended for logging purposes.
+     */
+    String getUnacknowledgedFlowfileInfo();
```

Comment:
also meant to say that returning a 'String' eliminates the caller/client from having any control on formatting and details of output.  It is fine to return the flow file objects *i think* since they're immutable by intent.